queries/c/indents.scm

```diff
@@ -1,28 +1,81 @@
 [
-  (compound_statement)
-  (preproc_arg)
+  (compound_statement) 
   (field_declaration_list)
   (case_statement)
   (enumerator_list)
-  (struct_specifier)
   (compound_literal_expression)
   (initializer_list)
-  (while_statement)
-  (for_statement)
-  (switch_statement)
+  (init_declarator)
   (expression_statement)
 ] @indent
-(if_statement condition: (_) @indent)
-((if_statement
-  consequence: (_) @_consequence
+
+(
+ if_statement
+  condition: (_) @indent
+)
+(
+  (if_statement
+    consequence: (_ ";" @indent_end) @_consequence
+  ) @indent
   (#not-has-type? @_consequence compound_statement)
-  ) @indent)
-(init_declarator) @indent
+) 
+
+;; if-else case,
+;; with else that is not if/compound statement
+(
+  (if_statement
+    consequence: (_ ";" @indent_end) @_consequence
+    "else" @branch
+    alternative: 
+      (_ ";" @indent_end) @_alternative 
+  )@indent
+  (#not-has-type? @_consequence compound_statement)
+  (#not-has-type? @_alternative if_statement compound_statement)
+)
+
+;; if- else case
+;; The alternative must be one of the 3 cases below
+;; because the alternative is a nested tree inside the original if
+;; dedent to offset the indent it should have gotten
+(
+  (if_statement
+    consequence: (_ ";" @indent_end) @_consequence
+    "else" @branch
+    alternative: 
+      [
+        (compound_statement) 
+        (if_statement) 
+        (if_statement consequence: (compound_statement) @dedent)
+      ] @dedent
+  )@indent
+  (#not-has-type? @_consequence compound_statement)
+)
+
+(
+  ERROR
+    "for" "(" @indent ";" ";" ")" @indent_end)
```

Comment:
This is to maintain similar behavior for `for` compared to current master branch behavior